runtime/shared_common/shrinit.cpp

```diff
@@ -3509,6 +3547,21 @@ j9shr_init(J9JavaVM *vm, UDATA loadFlags, UDATA* nonfatal)
 		config->jclURLCache = NULL;
 		config->jclURLHashTable = NULL;
 		config->jclUTF8HashTable = NULL;
+		if (J9_ARE_ALL_BITS_SET(runtimeFlags, J9SHR_RUNTIMEFLAG_ENABLE_CACHE_NON_BOOT_CLASSES)) {
+			config->romToRamHashTable = hashTableNew(OMRPORT_FROM_J9PORT(vm->portLibrary),
+					J9_GET_CALLSITE(), 0, sizeof(RomToRamEntry), sizeof(char *), 0,
+					J9MEM_CATEGORY_CLASSES, romToRamHashFn, romToRamEqualFn, NULL, NULL);
+			if (NULL == config->romToRamHashTable) {
+				SHRINIT_ERR_TRACE(verboseFlags, J9NLS_SHRC_SHRINIT_FAILURE_CREATE_ROMTORAMHASHTABLE);
+				goto _error;
+			}
+
+			if (omrthread_rwmutex_init(&(config->romToRamHashTableMutex), 0, "romToRamHashTable mutex")) {
```

Comment:
Please initialize the mutex first, then allocate the hashtable. If the hashtable allocation failed, the mutext should be deleted.

Reply:
Thanks! I forgot this from the last review.